### PR TITLE
fix(pins): #MED-310 fix pin create and edit

### PR DESCRIPTION
**Pull request by niko-amosse**
## Describe your changes

Block spaces before the title and description of a pin when creating or editing it. 
Allow setting the description to "" (empty).

## Checklist tests

check that it work (on ng2)

## Issue ticket number and link

[MED-310](https://jira.support-ent.fr/browse/MED-310)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: backend/src/main/java/fr/openent/mediacentre/service/impl/DefaultPinsService.java [hunk 1]
```diff
@@ -97,7 +97,7 @@ public Future<Optional<PinResource>> put(String idStructure, String idPin, JsonO
         JsonObject resourceUpdated = new JsonObject(); // update only title and description
         if (resource.containsKey(Field.PINNED_TITLE) && !resource.getString(Field.PINNED_TITLE).isEmpty())
             resourceUpdated.put(Field.PINNED_TITLE, resource.getString(Field.PINNED_TITLE));
-        if (resource.containsKey(Field.PINNED_DESCRIPTION) && !resource.getString(Field.PINNED_DESCRIPTION).isEmpty())
+        if (resource.containsKey(Field.PINNED_DESCRIPTION))
```
Comment (niko-amosse) [inline backend/src/main/java/fr/openent/mediacentre/service/impl/DefaultPinsService.java:100]: pour ne pas bloque un set de la description à "" (vide)

File: frontend/src/components/modals/create-pins/CreatePins.tsx [hunk 2]
```diff
@@ -114,7 +114,7 @@ export const CreatePins: React.FC<CreatePinsProps> = ({ refetch }) => {
                 size="md"
                 type="text"
                 value={title}
-                onChange={(e) => setTitle(e.target.value)}
+                onChange={(e) => setTitle(e.target.value.trimStart())}
```
Comment (niko-amosse) [inline frontend/src/components/modals/create-pins/CreatePins.tsx:117]: vire les espaces des premiers caractères

File: frontend/src/components/modals/create-pins/CreatePins.tsx [hunk 3]
```diff
@@ -129,7 +129,7 @@ export const CreatePins: React.FC<CreatePinsProps> = ({ refetch }) => {
                 size="md"
                 type="text"
                 value={description}
-                onChange={(e) => setDescription(e.target.value)}
+                onChange={(e) => setDescription(e.target.value.trimStart())}
```
Comment (niko-amosse) [inline frontend/src/components/modals/create-pins/CreatePins.tsx:132]: idem

File: frontend/src/components/modals/edit-pins/EditPins.tsx [hunk 4]
```diff
@@ -150,7 +150,12 @@ export const EditPins: React.FC<EditPinsProps> = ({ refetch }) => {
         <Button color="tertiary" onClick={handleCloseModal}>
           {t("mediacentre.cancel")}
         </Button>
-        <Button color="primary" type="submit" onClick={onSubmit}>
+        <Button
+          color="primary"
+          type="submit"
+          onClick={onSubmit}
+          disabled={title == ""}
```
Comment (ThomasSevagen) [inline frontend/src/components/modals/edit-pins/EditPins.tsx:157]: ES6 `title == ""` équivaut à `!title`

**Review by ThomasSevagen — APPROVED** [review]

File: frontend/src/components/modals/edit-pins/EditPins.tsx (hunk 4 above)
Comment (ThomasSevagen) [reply frontend/src/components/modals/edit-pins/EditPins.tsx:157]: et aussi on essaye de rester sur de l'égalité stricte `===` dans le sens où on utilise la surcouche Typescript, et si tu fais des égalités en bypassant l'égalité de type bah ç'est pas fou.
En clair tu peux totalement oublier les semi égalité et faire que des égalités strictes `===` `!==`

Comment (niko-amosse) [reply frontend/src/components/modals/edit-pins/EditPins.tsx]: corrigé

**Review by guillaume-roux — APPROVED** [review]

**Pull request merged**